wagtail/snippets/blocks.py

```diff
@@ -23,8 +24,15 @@ def target_model(self):
     def widget(self):
         from wagtail.snippets.widgets import AdminSnippetChooser
 
+        try:
+            icon = self.target_model.snippet_viewset.icon
+        except AttributeError as e:
+            raise ImproperlyConfigured(
+                f"Cannot use SnippetChooserBlock with non-snippet model {self.target_model}"
```

Comment:
> I wonder how people ended up using SnippetChooserBlock while not registering the model as a snippet.

Judging from @robmoorman's comment https://github.com/wagtail/wagtail/issues/11702#issuecomment-1992576175, it's not necessarily places where the snippet registration has been forgotten entirely, but rather where it's been made conditional on a config setting / feature flag.

I'm pretty sure that SnippetChooserBlock has never worked on non-snippet models - for example, the version from before you refactored the registration mechanism to be closer to ModelAdmin used to call [`get_snippet_model_from_url_params`](https://github.com/wagtail/wagtail/blob/00203c5a72b9fb1fdc0368a42a548197c50e211a/wagtail/snippets/views/snippets.py#L29-L42), which explicitly raised a 404 on non-snippet models. As such, I'm pretty confident that the occasions where this is coming up are indeed user error (and the correct remedy is to make the error message more informative), rather than an indication that this should be more permissive.